spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -347,12 +347,21 @@ class DeltaCatalog extends DelegatingCatalogExtension
       properties: util.Map[String, String]) : Table =
     recordFrameProfile("DeltaCatalog", "createTable") {
       if (DeltaSourceUtils.isDeltaDataSourceName(getProvider(properties))) {
+        val (props, writeOptions) = getTablePropsAndWriteOptions(properties)
+        writeOptions.foreach { case (k, v) =>
+          // Continue putting in Delta prefixed options to avoid breaking workloads
+          if (k.toLowerCase(Locale.ROOT).startsWith("delta.")) {
+            props.put(k, v)
+          }
```

Comment:
Why is this needed i.e. why do we need to put `options.delta.xyz` to delta table properties? In what scenarios workload can break?

Reply:
This is copied from the existing code: https://github.com/delta-io/delta/blob/d81ce5ee557ad9741052c9f3402fbef3ff91da87/spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala#L586-L588

I don't have enough context to answer this question... This is just for safety as before this change, all the Spark table properties are kept as the Delta table properties, including `options.xyz` and `options.delta.xyz`. After this change only `options.xyz` is excluded from Delta table properties. This is consistent with CTAS command.

Reply:
maybe I should respect that legacy config here as well

Reply:
+1 - just for consistency we can continue to do that.